.golangci.yml

```diff
@@ -0,0 +1,39 @@
+run:
+  timeout: 5m
+
+issues:
+  # Maximum issues count per one linter. Set to 0 to disable. Default is 50.
+  max-issues-per-linter: 0
+
+  # Maximum count of issues with the same text. Set to 0 to disable. Default is 3.
+  max-same-issues: 0
+
+  # Disable default exclude rules, see `golangci-lint run --help`.
+  # When true it disables revive's comment warnings, and we want them enabled.
+  exclude-use-default: false
+
+linters:
+  disable-all: true
+  enable:
+    - deadcode
+    - dupl
+    - gofmt
+    - goimports
+    - gosimple
+    - govet
+    - ineffassign
+    - misspell
+    - nakedret
+    - revive
+    - structcheck
+    - unused
+    - varcheck
+    - staticcheck
```

Comment:
Should we add more linters?

The [k6 config](https://github.com/grafana/k6/blob/0dda328d9a7f563607dae4fe89ff8c5138f8dc09/.golangci.yml) enables all and disables some, which I think is an error-prone approach, as new linters are constantly added, which often introduces new failures. So this is a variation of the [helm config](https://github.com/helm/helm/blob/35f6196167f6b90913f3a94237825856b858a215/.golangci.yml) which is a safer approach.

Reply:
Hey, I've summarized the other linters we might consider and the current number of warnings from them:

| linter | # of warnings |
| --- | --- |
| gocritic | 45 |
| errcheck | 3 |
| typecheck | 0 |
| cyclop | 37 |
| durationcheck | 1 |
| errchkjson | 6 |
| errorlint | 22 |
| exhaustive | 20 |
| exportloopref | 0 |
| forbidigo | 1 |
| funlen | 25 |
| gci | 33 |
| gochecknoglobals | 21 |
| gocognit | 11 |
| goconst | 33 |
| gocyclo | 1 |
| godot | 245 |
| gofumpt | 11 |
| gosec | 6 |
| ifshort | 6 |
| lll | 104 |
| makezero | 1 |
| misspell | 3 |
| nakedret | 0 |
| nestif | 11 |
| nilerr | 0 |
| nilnil | 4 |
| nlreturn | 435 |
| prealloc | 2 |
| thelper | 11 |
| tparallel | 6 |
| unconvert | 4 |
| unparam | 15 |
| wastedassign | 1 |
| whitespace | 4 |
| wrapcheck | 48 |
| forcetypeassert | 76 I'm not sure about this one. |

Should we enable `containedctx` or `ireturn` as well? Just kidding :)

Reply:
See: https://github.com/grafana/xk6-browser/pull/216#issuecomment-1029766734